### ISSUE TITLE
[Bug]: Unable to test module outside of GitHub

### ISSUE THREAD

**Issue by yibble** [purpleclay/daggerverse#23]
### What happened?

I have a module (`semantic-version`) that depends on the `nsv` module. But it's unable to call nsv when executed outside of a GitHub Workflow, for example, in local testing:

```
Stdout:
invoke: Next:build(deps).Error error: input: semanticVersion.next resolve: call function "Next": process "/runtime" did not complete successfully: exit code: 2

Stdout:
invoke: input: nsv resolve: call constructor: process "/runtime" did not complete successfully: exit code: 2

Stdout:
marshal: json: error calling MarshalJSON for type *main.Nsv: json: error calling MarshalJSON for type *dagger.Container: input: container.from resolve: failed to resolve image ghcr.io/purpleclay/nsv:v0.10.0: failed to resolve source metadata for ghcr.io/purpleclay/nsv:v0.10.0: failed to authorize: failed to fetch oauth token: unexpected status from GET request to https://ghcr.io/token?scope=repository%3Apurpleclay%2Fnsv%3Apull&service=ghcr.io: 403 Forbidden
```

### Steps to reproduce it

1.
2.
3.
...

### Which version?

v0.10.0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by purpleclay** [comment]
It is strange you are getting an auth error when the `nsv` project and its package registry are both public. I use this function within GitLab without issue. But I an invoking it directly.

Is your Dagger module public or private? And what version of the Dagger engine are you using?

**Comment by yibble** [comment]
It's a private Dagger module, and using v0.14.0. But I've experienced the same issue in earlier versions too.

**Comment by purpleclay** [comment]
I have created a branch and included a `testing` module that uses `nsv` as a dependency. When I call it, I do not get these auth issues, so it feels like a private module issue. I have to admit, I haven't used a private module yet. I see nothing that would prevent a pull from `ghcr.io` on my end.

```sh
dagger call -m github.com/purpleclay/daggerverse/testing@nsv-testing next --src .
```

**Comment by purpleclay** [comment]
> It's a private Dagger module, and using v0.14.0. But I've experienced the same issue in earlier versions too.

So does that mean it has never worked from a private module?

**Comment by yibble** [comment]
Yes, that's correct. It's never worked. I used to resolve this by keeping a local copy of your module, and then having a condition that if the release look-up failed for nsv, it would provide a hardcoded value.
